sdks/java/core/src/test/java/org/apache/beam/sdk/coders/CoderTest.java

```diff
@@ -74,4 +79,35 @@ public void testNonDeterministicException() {
     assertThat(exception.toString(), containsString("Problem"));
     assertThat(exception.toString(), containsString("is not deterministic"));
   }
+
+  @Test
+  public void testTypeIsPreserved() throws Exception {
```

Comment:
Is this because `VoidCoderTest` does not exist?

(tiny style nit: prefer `assertThat` and `equalTo` like you use elsewhere)